contracts/nois-gateway/src/contract.rs

```diff
@@ -505,6 +506,44 @@ fn execute_add_verified_round(
         .add_attributes(attributes))
 }
 
+fn execute_cleanup(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    limit: Option<u32>,
+) -> Result<Response, ContractError> {
+    // the number of elements we can still take (decreasing over time)
+    let mut limit = limit.unwrap_or(20) as usize;
+
```

Comment:
Does it need to ba a manager to execute this?

Reply:
No it's public. The ide is that whoever pays the gas for this cleanup is doing the chain a favour by removing unused state.

Reply:
Okay then I am missing the safety check where we filter on unused state.

Reply:
How do you know we're only removing the outdated keys?

Reply:
<img width="332" alt="Bildschirmfoto 2023-07-30 um 16 47 26" src="https://github.com/noislabs/nois-contracts/assets/2603011/3eebcb0e-933c-467f-9e41-c8e58dcd1f47">

Since 0.13.7 we read from both drand_jobs1 and drand_jobs2 but all new jobs go into drand_jobs2. drand_jobs1 is what this cleans up now.

Reply:
Okayy thank you now I understand. LGTM